### PR TITLE
feat(feedback): Sends MessageEvent when error page embed is closed

### PR DESCRIPTION
**Pull request by arya-s**
Sends `reportdialog_closed` via window.postMessage to the parent of the error page embed in preparation of supporting an `onClose` callback in @sentry/browser `showReportDialog`.

Fixes https://github.com/getsentry/sentry-javascript/issues/9433

Please see the accompanying PRs:
* for the sdk: https://github.com/getsentry/sentry-javascript/pull/9550
* for the docs: https://github.com/getsentry/sentry-docs/pull/8484

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

**Comment by arya-s** [comment]
> 🚨 **Warning:** This pull request contains Frontend and Backend changes!
> 
> It's discouraged to make changes to Sentry's Frontend and Backend in a single pull request. The Frontend and Backend are **not** atomically deployed. If the changes are interdependent of each other, they **must** be separated into two pull requests and be made forward or backwards compatible, such that the Backend or Frontend can be safely deployed independently.
> 
> Have questions? Please ask in the [`#discuss-dev-infra` channel](https://app.slack.com/client/T024ZCV9U/CTJL7358X).

I beleive this can be ignore, I'm only adding/changing tests here.

File: src/sentry/templates/sentry/error-page-embed.js [hunk 1]
```diff
@@ -146,11 +150,16 @@
       document.removeEventListener('keydown', handleFocus);
     }
     this.element.parentNode.removeChild(this.element);
+    if (window.parent) {
+      window.parent.postMessage('reportdialog_closed', '*');
```
Comment (AbhiPrasad) [inline src/sentry/templates/sentry/error-page-embed.js:154]:
```suggestion
      window.parent.postMessage('sentry_reportdialog_closed', '*');
```

Let's namespace to sentry to reduce possible collisions/confusions as much as possible

**Review by markstory — COMMENTED** [review]
Selenium tests and embed logic changes look good to me.

File: src/sentry/templates/sentry/error-page-embed.js (hunk 1 above)
Comment (arya-s) [reply src/sentry/templates/sentry/error-page-embed.js:154]: How about even more `__sentry_reportdialog_closed__`?

Comment (AbhiPrasad) [reply src/sentry/templates/sentry/error-page-embed.js]: Even better!

**Review by AbhiPrasad — APPROVED** [review]
Thanks a lot for the PR - lets get this deployed and then merge in changes for SDK + docs :)

**Comment by arya-s** [comment]
> Thanks a lot for the PR - lets get this deployed and then merge in changes for SDK + docs :)

My pleasure, please let me know if there's anything else needed.

**Pull request merged**